Mathlib/Analysis/NormedSpace/Star/Matrix.lean

```diff
@@ -70,3 +86,79 @@ theorem entrywise_sup_norm_bound_of_unitary {U : Matrix n n 𝕜} (hU : U ∈ Ma
 #align entrywise_sup_norm_bound_of_unitary entrywise_sup_norm_bound_of_unitary
 
 end EntrywiseSupNorm
+
+noncomputable section EuclideanSpaceOpNorm
+
+variable [Fintype n] [DecidableEq n] [IsROrC 𝕜]
+
+/-- The natural star algebra equivalence between matrices and continuous linear endomoporphisms
+of Euclidean space induced by the orthonormal basis `EuclideanSpace.basisFun`. -/
```

Comment:
```suggestion
/-- The natural star algebra equivalence between matrices and continuous linear endomoporphisms
of Euclidean space induced by the orthonormal basis `EuclideanSpace.basisFun`.

This is a more-bundled version of `Matrix.toEuclideanLin`, for the special case of square matrices. -/
```
Which of course raises the fact that we should probably have a version of this for rectangular matrices.

Reply:
I don't think we actually want a version of this for rectangular matrices, at least not as a separate `def`. That's because for rectangular matrices, using orthonormal bases doesn't really buy you any extra structure, at least not in the form of bundled maps. The best you can do is still `≃ₗ[𝕜]`, at which point you can simply write:
```lean
(Matrix.toEuclideanLin.trans LinearMap.toContinuousLinearMap : Matrix m n 𝕜 ≃ₗ[𝕜] (EuclideanSpace 𝕜 n →L[𝕜] EuclideanSpace 𝕜 m))
```
So it's just `LinearEquiv.trans` of two existing equivalences.

Reply:
It's true more generally in TVS, where `LinearMap.toContinuousLinearMap` isn't available.

Reply:
That just means we need to generalize `LinearMap.toContinuousLinearMap`?

Reply:
@eric-wieser now I'm really confused because I just looked at [LinearMap.toContinuousLinearMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Module/FiniteDimension.html#LinearMap.toContinuousLinearMap) and found that it already holds in a TVS. So what are you arguing for here?

Reply:
Sorry, I must have mis-remembered the generality of `LinearMap.toContinuousLinearMap`; indeed, my suggestion would be redundant here. Maybe there's some extra generality to eek out for `WithLp` over rings, but that's out of scope here.

Reply:
Can you still update the docstring to reference `Matrix.toEuclideanLin` as I suggest above? Adding a lemma making the link explicit would also be good.

Reply:
done and done